bot/exts/valentines/valentine_zodiac.py

```diff
@@ -29,13 +37,127 @@ def load_json() -> dict:
             zodiacs = load(json_data)
             return zodiacs
 
-    @commands.command(name="partnerzodiac")
-    async def counter_zodiac(self, ctx: commands.Context, zodiac_sign: str) -> None:
+    def zodiac_sign_verify(self, zodiac: str) -> discord.Embed:
+        """Gives informative zodiac embed."""
+        c_zodiac = zodiac.capitalize()
+        embed = discord.Embed()
+        embed.color = Colours.pink
+        if c_zodiac in zodiac_signs:
+            log.info("Making zodiac embed")
+            embed.title = f"__{c_zodiac}__"
+            embed.description = zodiac_fact[f"{c_zodiac}"]["About"]
+            embed.add_field(name='__Full form__', value=zodiac_fact[f"{c_zodiac}"]["full_form"], inline=False)
+            embed.add_field(name='__Motto__', value=zodiac_fact[f"{c_zodiac}"]["Motto"], inline=False)
+            embed.add_field(name='__Strengths__', value=zodiac_fact[f"{c_zodiac}"]["Strengths"], inline=False)
+            embed.add_field(name='__Weaknesses__', value=zodiac_fact[f"{c_zodiac}"]["Weaknesses"], inline=False)
+            embed.set_thumbnail(url=zodiac_fact[f"{c_zodiac}"]["url"])
+        else:
+            embed.description = "Umm you gave wrong zodiac name so i aren't able to find any :sweat_smile:"
+            log.info("Wrong Zodiac name provided")
+        return embed
+        log.info("Zodiac embed ready")
+
+    def zodiac_date_verifer(self, month: str, date: int) -> str:
+        """Returns zodiac sign by checking month and date."""
+        month = month.capitalize()
+        log.info("started searching zodaic sign based on month and date")
+        if month == "January" or month == "Jan":
```

Comment:
I suggest the following method for getting the zodiac sign as per month and date.
Have a DICT to store the starting and ending datetime objects of a zodiac sign
example:
`zodiacs = {
"Aquarius":(datetime(2020, 1, 20), datetime(2020, 2, 18))
}`
You can create a new datetime() object with the month and date passed by the user.
Then, you can check if that datetime() object is in between the starting and ending dates as specified in the dict.
lets call the datetime() object created with the user input *query_datetime*

`for zodiac_name, date_range in zodiacs.items(): 
    if (date_range[0]<=query_datetime<=date_range[1]):  
        # you found the zodiac`

Reply:
the starting and ending dates can also be presented in the JSON file, and the datetime() objects can be created during the Cog load.

Reply:
I think i should skip the year because year can change but date and month will remain same always

Reply:
The year can be anything as that wouldn't matter I believe